src/Components/Onboarding/Steps/Artists/ItemLink.tsx

```diff
@@ -62,16 +62,16 @@ class ItemLink extends React.Component<Props, State> {
   }
 
   followArtist(artist: ArtistFollowProps) {
-    const storeUpdater = (store: RecordSourceSelectorProxy) => {
-      const mutationPayload = store.getRootField("followArtist")
-      const origArtist = mutationPayload.getLinkedRecord("artistEdge")
-
-      // Useful to import this directly for debugging
-      const localConnHandler = ConnectionHandler
-
-      // Somehow update the artist with the suggestion...
-      const selectedArtist = store.get(artist.__id)
-      const connection = localConnHandler.getConnection(origArtist, "ItemLink_suggested")
+    const storeUpdater = (store: RecordSourceSelectorProxy, data: any) => {
+      // Search the store for a record for the newly suggested artist, based on its ID we get from the response `data`.
+      const suggestedArtist = store.get(data.followArtist.artistEdge.node.related.suggested.edges[0].node.__id)
```

Comment:
Maybe there’s a ‘nicer’ way to get the ID using the store API, but I don’t think it would be necessarily any more ‘correct’. The whole point of offering this imperative API is to be able to do things in the way people would more traditionally think about updating data after receiving a response.

I’ll leave it as an exercise to you to play with the dump/inspector debugger aids to figure out if there is a ‘nicer’ way at all, it’s probably worth it to spend 10 minutes playing with them to get a better understanding of the state of the in-memory store.

Reply:
So yea....exactly, I saw that I can just use the second argument, but I thought that wasn't the 'Relay' way so was struggling otherwise. 

I also was struggling with `store.get('client:root:..')`, try as I might, I simply could not figure that out from the documentation or any examples. I was also caught up in trying to do it the 'Relay' way and was trying to figure out how to make everything a connection, and attempt to swap out edges that way, rather than just doing it explicitly like this. Live and learn I guess.

Reply:
I actually would have kinda expected to be able to use `store.getRootField("popular_artists")`, but iirc that didn’t work. Maybe try again and otherwise file a ticket asking about that intended usage and if there’s a nicer ’Relay’ way to do all of this code. Then use that feedback to PR an improvement to the docs.

Reply:
A lil late but I'd recommend using `import { get } from "lodash"`
for `data.followArtist.artistEdge.node.related.suggested.edges[0].node.__id` to avoid an exception in case one of these properties is null or undefined

Reply:
cc @xtina-start